http/asio_thread_pool.cc

```diff
@@ -27,180 +28,47 @@ namespace Datacratic {
 /*****************************************************************************/
 
 struct AsioThreadPool::Impl {
-    Impl(boost::asio::io_service & ioService,
-         double probeIntervalSeconds = 0.1)
-        : ioService(ioService), shutdown(false)
+    Impl()
+        : loopCnt(0)
     {
-        work.reset(new boost::asio::io_service::work(ioService));
-
-        threads.emplace_back([=] () { this->run(0); });
-
-        lastProbe = Date::now();
-        lastLatency = 0;
-
-        timer = getTimer(Date::now().plusSeconds(probeIntervalSeconds),
-                         probeIntervalSeconds,
-                         ioService,
-                         std::bind(&Impl::onProbe,
-                                   this,
-                                   std::placeholders::_1));
-        numIdles = 0;
     }
 
     ~Impl()
     {
-        shutdown = true;
-        ioService.stop();
-        work.reset();
-        for (auto & t: threads) {
-            ioService.post([] () {});
-            t.join();
+        for (size_t i = 0; i < threads.size(); i++) {
+            loops[i]->terminate();
+            threads[i].join();
```

Comment:
Si je ne me trompe pas ça ne devrait pas être ici. thread.join n'a pas de "nothrow" alors ton dtor peut thrower. Not good.
http://en.cppreference.com/w/cpp/thread/thread/join